gapis/api/transform/dce.go

```diff
@@ -0,0 +1,230 @@
+package transform
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/gapid/core/app/benchmark"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/resolve/dependencygraph"
+)
+
+var (
+	dCECounter         = benchmark.GlobalCounters.Duration("DCE")
+	dCECmdDeadCounter  = benchmark.GlobalCounters.Integer("DCE.cmd.dead")
+	dCECmdLiveCounter  = benchmark.GlobalCounters.Integer("DCE.cmd.live")
+	dCEDrawDeadCounter = benchmark.GlobalCounters.Integer("DCE.draw.dead")
+	dCEDrawLiveCounter = benchmark.GlobalCounters.Integer("DCE.draw.live")
+	dCEDataDeadCounter = benchmark.GlobalCounters.Integer("DCE.data.dead")
+	dCEDataLiveCounter = benchmark.GlobalCounters.Integer("DCE.data.live")
+)
+
+type FullCommandIndexSet struct {
+	dependencygraph.ArrayBasedTrie
+}
+
+func newFullCommandIndexSet() *FullCommandIndexSet {
+	return &FullCommandIndexSet{*dependencygraph.NewArrayBasedTrie()}
+}
+
+func (s *FullCommandIndexSet) insert(fci dependencygraph.FullCommandIndex) {
+	s.SetValue(fci, true)
+}
+
+func (s *FullCommandIndexSet) contains(
+	fci dependencygraph.FullCommandIndex) bool {
+	v, ok := s.GetValue(fci)
+	if ok {
+		if bv, ok := v.(bool); ok {
+			return bv
+		}
+		return false
+	}
+	return false
+}
+
+type DefUseGraph struct {
```

Comment:
This type is here to distribute commands to their corresponding DefUse checking machine according to their API.